src/mame/video/k057714.cpp

```diff
@@ -368,33 +424,42 @@ void k057714_device::draw_frame(int frame, bitmap_ind16 &bitmap, const rectangle
 	if (m_frame[frame].x + width > cliprect.max_x)
 		width = cliprect.max_x - m_frame[frame].x;
 
-	for (int j = 0; j < height; j++)
+	for (int j = 0; j <= height; j++)
 	{
 		uint16_t *const d = &bitmap.pix(j + m_frame[frame].y, m_frame[frame].x);
 		int li = (j * fb_pitch);
-		for (int i = 0; i < width; i++)
+		for (int i = 0; i <= width; i++)
 		{
 			uint16_t pix = vram16[(m_frame[frame].base + li + i) ^ NATIVE_ENDIAN_VALUE_LE_BE(1, 0)];
 			if ((pix & 0x8000) != trans_value) {
-				d[i] = pix & 0x7fff;
+				uint32_t r = (pix >> 10) & 0x1f;
+				uint32_t g = (pix >> 5) & 0x1f;
+				uint32_t b = (pix >> 0) & 0x1f;
+
+				r = (r * alpha1) >> 4;
+				g = (g * alpha1) >> 4;
+				b = (b * alpha1) >> 4;
+
+				if (r > 0x1f) r = 0x1f;
+				if (g > 0x1f) g = 0x1f;
+				if (b > 0x1f) b = 0x1f;
+
+				d[i] = (r << 10) | (g << 5) | b;
 			}
 		}
 	}
 }
 
 int k057714_device::draw(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect)
 {
-	int width = m_frame[0].width;
-	int height = m_frame[0].height;
-
-	if (width != 0 && height != 0)
+	if (m_display_width != 0 && m_display_height != 0)
 	{
 		rectangle visarea = screen.visible_area();
-		if ((visarea.max_x+1) != width || (visarea.max_y+1) != height)
+		if (visarea.max_x != m_display_width || visarea.max_y != m_display_height)
 		{
-			visarea.max_x = width-1;
-			visarea.max_y = height-1;
-			screen.configure(width, height, visarea, screen.frame_period().attoseconds());
+			visarea.max_x = m_display_width;
+			visarea.max_y = m_display_height;
+			screen.configure(m_display_width, m_display_height, visarea, screen.frame_period().attoseconds());
```

Comment:
It would be better if the real screen timing parameters could be determined.  This is still pretending there’s no blanking area.